recipes/overlay-files/files/overlay.sh

```diff
@@ -18,7 +18,7 @@ for file in $OVERLAY_FILES ; do
     if test -e "$overlay_file" ; then
 	if test ! -h "$link_file" -o \
 	"`readlink "$link_file"`" != "$overlay_file" ; then
-	    ln -sf "$overlay_file" "$link_file"
+	    ln -sfn "$overlay_file" "$link_file"
```

Comment:
Why not use `-T` instead, which would cause a failure instead of a silent wrong action in case of $link_file being a plain directory?